mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/MqttHighLevelApi.scala

```diff
@@ -0,0 +1,246 @@
+) {
+
+  /** Scala API */
+  def withMinBackoff(value: scala.concurrent.duration.FiniteDuration): MqttRestartSettings = copy(minBackoff = value)
+
+  /** Java API */
+  def withMinBackoff(value: java.time.Duration): MqttRestartSettings = copy(minBackoff = value.asScala)
+
+  /** Scala API */
+  def withMaxBackoff(value: scala.concurrent.duration.FiniteDuration): MqttRestartSettings = copy(maxBackoff = value)
+
+  /** Java API */
+  def withMaxBackoff(value: java.time.Duration): MqttRestartSettings = copy(maxBackoff = value.asScala)
+  def withRandomFactor(value: Double): MqttRestartSettings = copy(randomFactor = value)
+  def withMaxRestarts(value: Int): MqttRestartSettings = copy(maxRestarts = value)
+
+  private def copy(
+      minBackoff: scala.concurrent.duration.FiniteDuration = minBackoff,
+      maxBackoff: scala.concurrent.duration.FiniteDuration = maxBackoff,
+      randomFactor: Double = randomFactor,
+      maxRestarts: Int = maxRestarts
+  ): MqttRestartSettings = new MqttRestartSettings(
+    minBackoff = minBackoff,
+    maxBackoff = maxBackoff,
+    randomFactor = randomFactor,
+    maxRestarts = maxRestarts
+  )
+
+  override def toString =
+    "MqttRestartSettings(" +
+    s"minBackoff=${minBackoff.toCoarsest}," +
+    s"maxBackoff=${maxBackoff.toCoarsest}," +
+    s"randomFactor=$randomFactor," +
+    s"maxRestarts=$maxRestarts" +
+    ")"
+}
+
+@ApiMayChange
+object MqttRestartSettings {
+
+  val Defaults = new MqttRestartSettings(
+    minBackoff = 2.seconds,
+    maxBackoff = 2.minutes,
+    randomFactor = 0.42d,
+    maxRestarts = -1
+  )
+
+  /** Scala API */
+  def apply(): MqttRestartSettings = Defaults
+
+  /** Java API */
+  def create(): MqttRestartSettings = Defaults
+
+}
+
+/**
+ * Create a `Subscribe` packet for high-level APIs.
+ * [[MqttSubscriptions]] is the most applicable implementation.
+ */
+@ApiMayChange
+trait MqttSubscribe {
+  def controlPacket: Subscribe
+}
+
+/**
+ * The mapping of topics to subscribe to and their control flags (including Quality of Service) per topic.
+ */
+@ApiMayChange
+final class MqttSubscriptions private (
+    val subscriptions: Map[String, ControlPacketFlags]
+) extends MqttSubscribe {
+
+  /** Scala API */
+  def withSubscriptions(subscriptions: Map[String, ControlPacketFlags]): MqttSubscriptions =
+    new MqttSubscriptions(subscriptions)
+
+  /** Java API */
+  def withSubscriptions(subscriptions: java.util.Map[String, ControlPacketFlags]): MqttSubscriptions =
+    new MqttSubscriptions(subscriptions.asScala.toMap)
+
+  /** Add this subscription to the map of subscriptions configured already. */
+  def addSubscription(topic: String, qos: ControlPacketFlags): MqttSubscriptions =
+    new MqttSubscriptions(this.subscriptions.updated(topic, qos))
+
+  def addAtLeastOnce(topic: String): MqttSubscriptions =
+    new MqttSubscriptions(subscriptions.updated(topic, ControlPacketFlags.QoSAtLeastOnceDelivery))
+
+  def controlPacket: Subscribe = Subscribe(subscriptions.toList)
```

Comment:
Other QoS levels?

Reply:
There is `addSubscription`.

Reply:
Ah yes. Thanks.

Reply:
I guess for consistency though you could have `addAtMostOnce` and `addExactlyOnce`...?